File: routes/overflows.js
const express = require('express');
const { getOverflows, getOverflow, createOverflow, commentOnOverflow, getUserOverflows } = require('../controllers/overflowController');
const auth = require('../middlewares/auth');

const router = express.Router();

router.get('/', getOverflows);
router.get('/:id', getOverflow);
router.get('/user/:username', getUserOverflows); // Changed to '/user/:username'
router.post('/', auth, createOverflow);
router.post('/:id/comments', auth, commentOnOverflow);

module.exports = router;

File: routes/overflows.go
package routes

import (
	"github.com/gin-gonic/gin"
	"Backend/controllers"
	"Backend/middlewares"
)

func OverflowRoutes(router *gin.RouterGroup) {
	router.GET("/", controllers.GetOverflows)
	router.GET("/:id", controllers.GetOverflow)
	router.GET("/user/:username", controllers.GetUserOverflows)
	router.POST("/", middlewares.Auth(), controllers.CreateOverflow)
	router.POST("/:id/comments", middlewares.Auth(), controllers.CommentOnOverflow)
}
